src/v2i-hub/RSUHealthMonitorPlugin/src/RSUHealthMonitorPlugin.cpp

```diff
@@ -10,49 +10,44 @@ namespace RSUHealthMonitor
         _rsuStatusTimer = make_unique<ThreadTimer>();
+        _rsuConfigListPtr = std::make_shared<RSUConfigurationList>();
         UpdateConfigSettings();
 
         // Send SNMP call to RSU periodically at configurable interval.
         _timerThId = _rsuStatusTimer->AddPeriodicTick([this]()
                                                       {
-            // Periodic SNMP call to get RSU status based on RSU MIB version 4.1
-            auto rsuStatusJson =  _rsuWorker->getRSUStatus(_rsuMibVersion, _rsuIp, _snmpPort, _securityUser, _authPassPhrase, _securityLevel, SEC_TO_MICRO);
-            PLOG(logINFO) << "Updating _interval: " << _interval;
-            //Broadcast RSU status periodically at _interval
-            BroadcastRSUStatus(rsuStatusJson); },
+                        this->monitorRSUs();
+                        PLOG(logINFO) << "Monitoring RSU at interval (second): " << _interval; },
                                                       std::chrono::milliseconds(_interval * SEC_TO_MILLI));
         _rsuStatusTimer->Start();
     }
 
+    void RSUHealthMonitorPlugin::monitorRSUs()
+    {
+        for (auto rsuConfig : _rsuConfigListPtr->getConfigs())
+        {
+            auto rsuStatusJson = _rsuWorker->getRSUStatus(rsuConfig.mibVersion, rsuConfig.rsuIp, rsuConfig.snmpPort, rsuConfig.user, rsuConfig.authPassPhrase, rsuConfig.securityLevel, SEC_TO_MICRO);
+            BroadcastRSUStatus(rsuStatusJson, rsuConfig.mibVersion);
+        }
+    }
+
     void RSUHealthMonitorPlugin::UpdateConfigSettings()
     {
         PLOG(logINFO) << "Updating configuration settings.";
 
         lock_guard<mutex> lock(_configMutex);
         GetConfigValue<uint16_t>("Interval", _interval);
-        GetConfigValue<string>("RSUIp", _rsuIp);
-        GetConfigValue<uint16_t>("SNMPPort", _snmpPort);
-        GetConfigValue<string>("AuthPassPhrase", _authPassPhrase);
-        GetConfigValue<string>("SecurityUser", _securityUser);
-        GetConfigValue<string>("SecurityLevel", _securityLevel);
-        GetConfigValue<string>("RSUMIBVersion", _rsuMIBVersionStr);
-        boost::trim_left(_rsuMIBVersionStr);
-        boost::trim_right(_rsuMIBVersionStr);
-        // Support RSU MIB version 4.1
-        if (boost::iequals(_rsuMIBVersionStr, RSU4_1_str))
-        {
-            _rsuMibVersion = RSUMibVersion::RSUMIB_V_4_1;
-        }
-        else
-        {
-            _rsuMibVersion = RSUMibVersion::UNKOWN_MIB_V;
-            PLOG(logERROR) << "Uknown RSU MIB version: " << _rsuMIBVersionStr;
-        }
+        GetConfigValue<string>("RSUConfigurationList", _rsuConfigListStr);
 
         try
         {
+            _rsuConfigListPtr->parseRSUs(_rsuConfigListStr);
             _rsuStatusTimer->ChangeFrequency(_timerThId, std::chrono::milliseconds(_interval * SEC_TO_MILLI));
         }
+        catch (const RSUConfigurationException &ex)
+        {
+            PLOG(logERROR) << "Cannot update RSU configurations due to error: " << ex.what();
```

Comment:
What do we want to happen when there is an error? Will this clear the previous list?

Reply:
It will use the previous list. No clear. In the parseRSUS() function, I left a comment saying that it will not clear the list until all configs are updated correctly.